SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialMetricsDisplay.cpp

```diff
@@ -181,10 +193,39 @@ void ASpatialMetricsDisplay::Tick(float DeltaSeconds)
 
 	const USpatialMetrics& Metrics = *SpatialNetDriver->SpatialMetrics;
 
-	FWorkerStats Stats;
+	FWorkerStats Stats{};
 	Stats.WorkerName = SpatialNetDriver->Connection->GetWorkerId().Left(WorkerNameMaxLength).ToLower();
 	Stats.AverageFPS = Metrics.GetAverageFPS();
-	Stats.WorkerLoad = Metrics.GetWorkerLoad();
+
+#if USE_SERVER_PERF_COUNTERS
+	float MovementCorrectionsPerSecond = 0.f;
+	int32 NumServerMoveCorrections = 0;
+	float WorldTime = GetWorld()->GetTimeSeconds();
+	NumServerMoveCorrections = PerfCountersGet(TEXT("NumServerMoveCorrections"), NumServerMoveCorrections);
+	MovementCorrectionRecord OldestRecord;
+	if (MovementCorrectionRecords.Peek(OldestRecord))
+	{
+		const float WorldTimeDelta = WorldTime - OldestRecord.Time;
+		const int32 CorrectionsDelta = NumServerMoveCorrections - OldestRecord.MovementCorrections;
+		if (WorldTimeDelta > 0.f && CorrectionsDelta > 0)
+		{
+			MovementCorrectionsPerSecond = CorrectionsDelta / WorldTimeDelta;
+		}
+
+		// Store the most recent 30 seconds of game time worth of measurements
+		if (WorldTimeDelta > 30.f)
+		{
+			MovementCorrectionRecords.Pop();
+		}
+	}
+	Stats.ServerMovementCorrections = MovementCorrectionsPerSecond;
```

Comment:
Is per-second not too volatile? Wouldn't it be better to do per minute? Also, is the overhead of pushing a metric on every tick worth it? Maybe we can do less often and get the same results.

Reply:
It's a 30 second sliding window. Can up it to a minute if you prefer. Testing this with 10 simulated players in the AIGym, it sat around 1-2.

Reply:
Ok let's leave it like this. That number sounds a bit off to me but let's see this in a real playtest and ensure it correlates well to perceived performance.